File: LeetCode/Medium/0334-increasing-triplet-subsequence/0334-increasing-triplet-subsequence.go
func increasingTriplet(nums []int) bool {
    first := 2147483647
    second := 2147483647
    for i := 0; i < len(nums); i++ {
        if nums[i] <= first {
            first = nums[i]
        } else if nums[i] <= second {
            second = nums[i]
        } else {
            return true
        }
    }
    return false
}

File: LeetCode/Medium/0334-increasing-triplet-subsequence/0334-increasing-triplet-subsequence.cpp
class Solution {
public:
    bool increasingTriplet(vector<int>& nums) {
        int first = INT_MAX;
        int second = INT_MAX;
        int temp;
        for (int i = 0; i < nums.size(); ++i){
            temp = nums[i];
            if (first >= temp) first = temp;
            else if (second >= temp) second = temp;
            else return true;
        }
        return false;
    }
};